lib/public/views/Environments/ActiveColumns/environmentsActiveColumns.js

```diff
@@ -32,13 +33,16 @@ export const environmentsActiveColumns = {
         size: 'w-10 f6 w-wrapped',
         visible: true,
         primary: true,
-        format: (environmentId) => buttonLinkWithDropdown(
+        format: (environmentId, { status }) => buttonLinkWithDropdown(
             environmentId,
             'env-details',
             { environmentId },
             [
                 h(CopyToClipboardComponent, { value: environmentId, id: environmentId }, 'Copy Environment Id'),
                 ...infologgerLinksComponents({ environmentId }),
+                status === 'RUNNING'
+                    ? getAliEcsEnvironmentLink(environmentId)
+                    : null,
```

Comment:
I have put the logic of either the link should exist or not in QCG url provider function, not in the component, please do the same.
If you think this is a bad idea, feel free to explain

Reply:
applied

Reply:
It do not looks like it?

Reply:
ok, I got what you meant, no, it won't be as there are different decision criteria for environments and runs also runs don't include environment entity so cannot do it one way